### PR TITLE
Use Github Actions to run tests

### PR DESCRIPTION
**Pull request by dolfinus**
Just like https://github.com/duffelhq/paginator/pull/90, but instead of CircleCI it's better to use Github Actions.
Github Actions supports matrix builds (unlike Circle CI), and also allows to build really impressive CI/CD pipelines and PR/issues automatization. [Here](https://docs.github.com/en/actions/learn-github-actions/managing-complex-workflows) you can find more details about features.
Projects like [ecto](https://github.com/elixir-ecto/ecto/pull/3118) and [phoenix](https://github.com/phoenixframework/phoenix/pull/3557) are already moved to Github Actions.

Here you can find an example of successfully executed workflow: https://github.com/dolfinus/paginator/actions/runs/610852525

### CODE REVIEW

**Comment by dolfinus** [comment]
@sgerrand @jbernardo95 @msz What do you think about this PR?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          ${{ runner.os }}-build-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
+          ${{ runner.os }}-build-${{ matrix.otp }}-${{ matrix.elixir }}
+          ${{ runner.os }}-build-${{ matrix.otp }}
+          ${{ runner.os }}-build
+
+    - name: Install Dependencies
+      run: |
+        mix local.hex --force
+        mix local.rebar --force
+        mix deps.get
```
Comment (sgerrand) [inline .github/workflows/test.yml:86]: Suggest running these as separate steps. There's little runtime cost to doing so and it makes debugging easier.
```suggestion
    - name: Install hex
      run: mix local.hex --force

    - name: Install rebar
      run: mix local.rebar --force

    - name: Install package dependencies
      run: mix deps.get
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    - name: Run unit tests
+      run: |
+        mix clean
+        mix test
```
Comment (sgerrand) [inline .github/workflows/test.yml:91]: Again, suggest separating these commands into separate steps.
```suggestion
    - name: Remove compiled application files
      run: mix clean

    - name: Run unit tests
      run: mix test
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    - name: Check documentation quality
+      run: |
+        mix inch.report
```
Comment (sgerrand) [inline .github/workflows/test.yml:95]: A tiny nitpick here
```suggestion
    - name: Check documentation quality
      run: mix inch.report
```

**Review by sgerrand — CHANGES_REQUESTED** [review]
Thank you for your contribution! These changes look good overall. I've added a few suggestions.

One thing missing is a link to this workflow running successfully. Would you please add this, either in the description or as a comment, to make it easier for other reviewers.

**Comment by sgerrand** [comment]
I think one other thing which needs to be done before this can be merged is enabling GitHub Actions on this repository and ensuring that at least 2-3 runs of this workflow complete successfully.

**Comment by sgerrand** [comment]
> Github Actions supports matrix builds (unlike Circle CI)

Note that CircleCI has [supported matrices in their builds](https://circleci.com/blog/circleci-matrix-jobs/) since April 2020. I only became aware of this myself late last year.

**Comment by dolfinus** [comment]
Thank your for review!

I've added a link to a successfully executed workflow.

**Review by sgerrand — APPROVED** [review]
LGTM. 👍

**Pull request merged**